enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,372 @@
+
+In the following proposal, the controller supports mainly 
+1. Day 2 operations and tag management for cloud resources selected using classifiers.
+2. Own the tag list and reconcile to replace any edits performed external to the controller.
+3. Opcodes in specification for handling user request as an operation on tag list.
+4. Extensible APIs and segregated configuration for controller and cloud provider specific configurations.
+
+## Motivation
+
+Users should be able to maintain lifecycle of tags for cloud resources created during installation or by other day-2 kubernetes
+controllers and operators using kubernetes API.
+
+### User stories
+
+As a cluster admin, I want to add tags to the cloud resources created at cluster installation,  which help
+for cost calculation and reports.
+
+As a cluster admin, I want tags to be added to the cloud resources created post cluster installation which help to include
+cloud resources for other tag processors.
+
+As a cluster admin, I want all tags to be added via metadata CR, which helps to reconcile the tag key and values.
+
+As a cluster admin, I want to delete tags, which are not required on the cloud resources.
+
+As a cluster admin, I want to add tags on selected resources, which are identified using specified labels/classifiers.
+
+As a cluster admin, I want to be able to ignore cloud resource from all tag operations, which helps to avoid overwriting values 
+set by cloud service provider's policies.
+
+As a cluster admin, I want the tag list to be lexicographically sorted, to have a deterministic set of tags applied 
+when tag list size exceeds the maximum allowed limit on cloud resource.
+
+As a cluster admin, I want the tag list to be lexicographically sorted, to have a deterministic set of tags appended to 
+existing tags for cloud resources when the tag list exceeds the maximum allowed limit on cloud resource.
+
+As a cluster admin, I want to use a CLI, that helps to securely login to cloud service provider and perform configuration and management operations for tags.
+
+### Goals
+
+1. Enable continuous management operations (create, append , update and delete) applicable for tagging 
+ cloud resources created by kubernetes services.
+2. Provide a new custom resource of kubernetes api for specification and status of tags.
+3. Enable continuous monitoring of tags applied and reconcile when there are changes made external to 
+the controller's API.
+4. Provide CLI for operations, cloud service provide authentication and ability to edit custom resource with ease.
+
+### Non-goals
+
+1. Managing metadata which are not related to cloud resource tagging.
+2. Data-related metadata for stored objects and records, for example, S3 bucket vs S3 object.
+3. Sub-set of input tags to be applied on selected cloud resource.
+4. Managing conflict with cloud service provider tagging policies.
+5. Preservation of pre-existing tags or tags added using external tools.
```

Comment:
i think we've previously said we wouldn't touch existing tags on resources for fear of removing a tag a customer added manually.  Is this moving away from that position?